src/bigint/add.rs

```diff
@@ -86,6 +87,149 @@ impl<const N_BITS: u32, const LIMB_SIZE: u32> BigIntImpl<N_BITS, LIMB_SIZE> {
             }
         }
     }
+
+    // double the item on top of the stack
+    pub fn double_allow_overflow() -> Script {
+        script! {
+            { 1 << LIMB_SIZE }
+
+            // Double the limb, take the result to the alt stack, and add initial carry
+            limb_double_without_carry OP_TOALTSTACK
+
+
+            for _ in 0..Self::N_LIMBS - 2 {
+                limb_double_with_carry OP_TOALTSTACK
+            }
+
+            // When we got {limb} {base} {carry} on the stack, we drop the base
+            OP_NIP // {limb} {carry}
+            { limb_double_with_carry_allow_overflow(Self::HEAD_OFFSET) }
+
+            // Take all limbs from the alt stack to the main stack
+            for _ in 0..Self::N_LIMBS - 1 {
+                OP_FROMALTSTACK
+            }
+        }
+    }
+
+    // double the item on top of the stack
```

Comment:
```suggestion
    /// Double the item on top of the stack
    /// 
    /// # Note
    ///
    /// This function prevents overflow of the underlying integer types during
    /// doubling operation.
```
Please as the other docstring change if necessary.